pokemongo_bot/event_handlers/telegram_handler.py

```diff
@@ -21,10 +21,22 @@ def handle_event(self, event, sender, level, formatted_msg, data):
             if event == 'level_up':
                 msg = "level up ({})".format(data["current_level"])
             elif event == 'pokemon_caught':
-                if data["pokemon"] in self.pokemons or self.pokemons[0]=="all":
-                    msg = "Caught {} CP: {}, IV: {}".format(data["pokemon"],data["cp"],data["iv"])
+                if isinstance(self.pokemons, list):
+                    if data["pokemon"] in self.pokemons or "all" in self.pokemons:
+                        msg = "Caught {} CP: {}, IV: {}".format(data["pokemon"],data["cp"],data["iv"])
+                    else:
+                        return
                 else:
-                    return
+                    if data["pokemon"] in self.pokemons:
+                        trigger = self.pokemons[data["pokemon"]]
+                    elif "all" in self.pokemons:
+                        trigger = self.pokemons["all"]
+                    else:
+                        return
+                    if (not "operator" in trigger or trigger["operator"] == "and") and data["cp"] >= trigger["cp"] and data["iv"] >= trigger["iv"] or ("operator" in trigger and trigger["operator"] == "or" and (data["cp"] >= trigger["cp"] or data["iv"] >= trigger["iv"])):
```

Comment:
There's good logic operator code in the pokemon catch worker, would you like to use that code @DBa2016